cmd/xl-storage-format-v2.go

```diff
@@ -510,6 +510,9 @@ func (z *xlMetaV2) DeleteVersion(fi FileInfo) (string, bool, error) {
 				if updateVersion {
 					delete(z.Versions[i].DeleteMarker.MetaSys, xhttp.AmzBucketReplicationStatus)
 					delete(z.Versions[i].DeleteMarker.MetaSys, VersionPurgeStatusKey)
+					if z.Versions[i].DeleteMarker.MetaSys == nil {
+						z.Versions[i].DeleteMarker.MetaSys = make(map[string][]byte)
+					}
```

Comment:
do you not want to add the code in a slightly different manner? 

```suggestion
					if len(z.Versions[i].DeleteMarker.MetaSys) == nil {
                                                 z.Versions[i].DeleteMarker.MetaSys = make(map[string][]byte)
                                        }
					delete(z.Versions[i].DeleteMarker.MetaSys, xhttp.AmzBucketReplicationStatus)
					delete(z.Versions[i].DeleteMarker.MetaSys, VersionPurgeStatusKey)
```

More safe this way and readable